launch_system/elasticache.tf

```
@@ -0,0 +1,43 @@
+resource "aws_security_group" "redis" {
+  name        = "launch_system_redis_sg"
+  description = "Security group for Redis instance"
+
+  vpc_id = var.vpc_id
+
+  ingress {
+    from_port   = 6379
+    to_port     = 6379
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr_block]
+    description = "allow access from within VPC"
+  }
+
+  tags = merge(var.tags, { Name = "launch_system_redis" })
+}
+
+# Create ElastiCache subnet group with private subnets
+resource "aws_elasticache_subnet_group" "redis" {
+  name = "launch_system_redis_subnet_group"
```

```suggestion
  name = "launch-system-redis-subnet-group"
```

When deploying manually I saw this error; this change should fix it: 

```
│ Error: creating ElastiCache Subnet Group (launch_system_redis_subnet_group): operation error ElastiCache: CreateCacheSubnetGroup, https response error StatusCode: 400, RequestID: 039acfd9-143c-473a-82ee-27307eb959ae, api error InvalidParameterValue: The parameter CacheSubnetGroupName is not a valid identifier. Ide
ntifiers must begin with a letter; must contain only ASCII letters, digits, and hyphens; and must not end with a hyphen or contain two consecutive hyphens.
```

---

Fixed in the last commit, thanks for spotting the error.
I find quite annoying that some names and ids don't allow the underscore as a valid character, while others allow it. It seems inconsistent, do you know if there is a clear rule?

---

Unfortunately "it's inconsistent" is as far as I got as well.